### PR TITLE
Tests for CASSANDRA-7392

### PR DESCRIPTION
**Pull request by stef1927**
Some tests to check that we can abort queries when they are already in progress, CASSANDRA-7392.

### CODE REVIEW

File: cql_tests.py [hunk 1]
```diff
@@ -394,3 +394,96 @@ def range_slice_test(self):
...
+@since('3.0')
+@require("7392")
+class AbortedQueriesTester(CQLTester):
+    """
+    @jira_ticket CASSANDRA-7392
+    Test that we can abort long queries that are already in progress
+    """
+    def local_query_test(self):
+        """
+        Abort a query running on the local coordinator node
```
Comment (mambocab) [inline cql_tests.py]: This docstring implies (to me) that this tests the correctness of some command that aborts the query. Upon further inspection, though, it looks like the query aborts because it takes longer than 1 second? If that's the case, maybe rephrase this to something like "Test that reads that take longer than `read_request_timeout_in_ms` time out".

File: cql_tests.py [hunk 2]
```diff
@@ -394,3 +394,96 @@ def range_slice_test(self):
...
+        """
+        cluster = self.cluster
+        cluster.set_configuration_options(values={'read_request_timeout_in_ms': 1000})
+
+        cluster.populate(1).start(wait_for_binary_proto=True, jvm_args=["-Dcassandra.test.read_iteration_delay_ms=100"])
```
Comment (mambocab) [inline cql_tests.py]: I don't know what this JVM argument does; could you please add a comment to explain?

File: cql_tests.py [hunk 3]
```diff
@@ -394,3 +394,96 @@ def range_slice_test(self):
...
+                id int PRIMARY KEY,
+                col int,
+                val text
+            );
+        """)
+
+        cursor.execute("CREATE INDEX ON test3 (col)")
+
+        for i in xrange(500):
+            cursor.execute("INSERT INTO test3 (id, col, val) VALUES ({}, {}, 'foo')".format(i, i / 10))
```
Comment (mambocab) [inline cql_tests.py]: We plan to transition to Python 3 in the near future. When we do, the return type of `/` will change from `int` to `float`:

```python
$ python -c 'print type(3/1)'
<type 'int'>
$ python3 -c 'print(type(3/1))'
<class 'float'>
```

It looks like you actually want floor division here, so could you please change this to `i // 10` so the behavior will be consistent between Python 2 and 3? I missed this incompatibility in `CONTRIBUTING.md`; I'll add that now.

**Comment by mambocab** [comment]
If it's worth testing that queries on indexes time out properly, it's probably also worth checking that it works correctly on materialized views. If you agree, and if you can add a test for that at this point, could you please do so with `@require('6477')`? If you're not able right now, this should be on @aboudreault's radar so he can implement the test (see #360).

**Comment by stef1927** [comment]
Thanks for the comments, see second commit. 

I cannot easily create a materialized view test right now as it would be too time consuming to rebase the 7392 branch on to the 6477 branch. Later, when both features are on the same branch (trunk, 6477 or 7392 - whichever comes first) we can then write a test. I've added a stub with a @require(6477) that fails so we don't forget to add this test once MV are committed.

**Comment by mambocab** [comment]
Perfect, thanks! Merging.
